akka-cluster-tools/src/multi-jvm/scala/akka/cluster/singleton/TeamSingletonManagerSpec.scala

```diff
@@ -0,0 +1,124 @@
+    akka.actor.serialize-creators = off
+    akka.remote.log-remote-lifecycle-events = off"""))
+
+  nodeConfig(controller)(
+    ConfigFactory.parseString("akka.cluster.team = one\n" +
+      "akka.cluster.roles = [ ]"))
+  nodeConfig(first) {
+    ConfigFactory.parseString("akka.cluster.team = one\n" +
+      "akka.cluster.roles = [ worker ]")
+  }
+  nodeConfig(second)(
+    ConfigFactory.parseString("akka.cluster.team = two\n" +
+      "akka.cluster.roles = [ worker ]"))
+  nodeConfig(third)(
+    ConfigFactory.parseString("akka.cluster.team = two\n" +
+      "akka.cluster.roles = [ worker ]"))
+}
+
+class TeamSingletonManagerMultiJvmNode1 extends TeamSingletonManagerSpec
+class TeamSingletonManagerMultiJvmNode2 extends TeamSingletonManagerSpec
+class TeamSingletonManagerMultiJvmNode3 extends TeamSingletonManagerSpec
+class TeamSingletonManagerMultiJvmNode4 extends TeamSingletonManagerSpec
+
+class TeamSingleton extends Actor with ActorLogging {
+  import TeamSingleton._
+
+  val cluster = Cluster(context.system)
+
+  override def receive: Receive = {
+    case Ping ⇒
+      sender() ! Pong(cluster.settings.Team, cluster.selfAddress, cluster.selfRoles)
+  }
+}
+object TeamSingleton {
+  case object Ping
+  case class Pong(fromTeam: String, fromAddress: Address, roles: Set[String])
+}
+
+abstract class TeamSingletonManagerSpec extends MultiNodeSpec(TeamSingletonManagerSpec) with STMultiNodeSpec with ImplicitSender {
+  import TeamSingletonManagerSpec._
+
+  override def initialParticipants = roles.size
+
+  val cluster = Cluster(system)
+  cluster.join(node(controller).address)
+  enterBarrier("nodes-joined")
+
+  val worker = "worker"
+
+  "A SingletonManager in a team" must {
+    "start a singleton instance for each team" in {
+
+      runOn(first, second, third) {
+        system.actorOf(
+          ClusterSingletonManager.props(
+            Props[TeamSingleton](),
+            PoisonPill,
+            ClusterSingletonManagerSettings(system).withRole(worker)),
+          "singletonManager")
+      }
+
+      val proxy = system.actorOf(ClusterSingletonProxy.props(
+        "/user/singletonManager",
+        ClusterSingletonProxySettings(system).withRole(worker)))
+
+      enterBarrier("managers-started")
+
+      proxy ! TeamSingleton.Ping
+      val pong = expectMsgType[TeamSingleton.Pong](10.seconds)
+
+      enterBarrier("pongs-received")
+
+      pong.fromTeam should equal(Cluster(system).settings.Team)
+      pong.roles should contain(worker)
+      runOn(controller, first) {
+        pong.roles should contain("team-one")
+      }
+      runOn(second, third) {
+        pong.roles should contain("team-two")
+      }
+
+      enterBarrier("after-1")
+    }
+
+    "be able to use proxy across different team" in {
+      runOn(third) {
+        val proxy = system.actorOf(ClusterSingletonProxy.props(
+          "/user/singletonManager",
+          ClusterSingletonProxySettings(system).withRole(worker).withTeam("one")))
```

Comment:
Normally you'd configure the proxy and the manager with the same role/team settings, right? I get that this works and demonstrates the desired behaviour, but a completely separate testcase for the situation where a singleton is restricted to a single team might be more realistic and serve as better 'documentation'?